elements/pl-file-preview/pl-file-preview.mustache

```diff
@@ -41,3 +39,49 @@
     </div>
   </div>
 </div>
+
+<script>
+$(() => {
+  function decodeBase64(str) {
+    return decodeURIComponent(Array.prototype.map.call(atob(str), (c) => {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
+    }).join(''));
+  }
+
+  function downloadFile(name, contents) {
+    const aElement = document.createElement('a');
+    aElement.setAttribute('href', 'data:application/octet-stream;base64,' + contents);
+    aElement.setAttribute('download', name);
+    aElement.setAttribute('target', '_blank');
+    aElement.click();
+  }
+
+  document.querySelectorAll('#file-preview-{{uuid}}').forEach((el) => {
+    el.querySelectorAll('.list-group-item').forEach((li) => {
+      const name = li.getAttribute('data-name');
+      const contents = li.getAttribute('data-contents')
+      const hasDecodeError = li.getAttribute('data-decode-error') === 'true';
```

Comment:
Personally I prefer using [`dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) instead, it's clearer:
```suggestion
      const name = li.dataset.name;
      const contents = li.dataset.contents;
      const hasDecodeError = li.dataset.decodeError === 'true';
```